trivup/apps/OauthbearerOIDCApp.py

```diff
@@ -0,0 +1,221 @@
+#   and/or other materials provided with the distribution.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED.
+# IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+# FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+# DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+# SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+# OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+from trivup import trivup
+from http.server import BaseHTTPRequestHandler, HTTPServer
+import jwt
+import datetime
+import json
+import argparse
+import requests
+from Crypto.PublicKey import RSA
+
+public_keys = []
+
+
+class WebServerHandler(BaseHTTPRequestHandler):
+    def do_GET(self):
+        self.send_response(200)
+        self.send_header('Content-type', 'text')
+        self.end_headers()
+        message = "HTTP server for OAuth\n"
+        message += "Example for token retrieval:\n"
+        message += 'curl \
+        -X POST \
+        --url localhost:8000/retrieve \
+        -H "Accept: application/json" \
+        -H "Authorization: Basic LW4gYWJjMTIzOlMzY3IzdCEK" \
+        -H "Cache-Control: no-cache" \
+        -d "method=oidc,scope=test-scope"'
+        self.wfile.write(message.encode())
+        print(message)
+
+    def generate_valid_token_for_client(self):
+        """
+        Example usage:
+        curl \
+        -X POST \
+        --url localhost:8000/retrieve \
+        -H "Accept: application/json" \
+        -H "Authorization: Basic LW4gYWJjMTIzOlMzY3IzdCEK" \
+        -H "Cache-Control: no-cache" \
+        -d "method=oidc,scope=test-scope"
+        """
+        if self.headers.get('Content-Length', None) is None:
+            self.send_error(404, 'Content-Length field is required')
+            return
+
+        content_length = int(self.headers['Content-Length'])
+        post_data = self.rfile.read(content_length)
+
+        if self.headers.get('Cache-Control', None) != "no-cache":
+            self.send_error(404, 'Cache-Control should be "no-cache"')
+            return
+
+        if self.headers.get('Authorization', None) is None:
+            self.send_error(404, 'Authorization field is required')
+            return
+
+        if self.headers.get('Accept', None) != "application/json":
+            self.send_error(404, 'Accept field should be "application/json"')
+            return
+
+        if post_data is None:
+            self.send_error(404,
+                            'method and scope fields are required in data')
```

Comment:
Per the above, for my own edification, can you point me to where the `method` parameter is described in the spec or on the Internet? My Google/DDG-fu isn't finding it. It would be helpful so I can update the Java client.

Reply:
Hi Kirk, never mind for this one, I will update it to grant_type=client_credentials